infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/UpgradeManager.java

```diff
@@ -0,0 +1,207 @@
+package org.corfudb.infrastructure.logreplication.utils;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.logreplication.infrastructure.LRRollingUpgradeHandler;
+import org.corfudb.infrastructure.logreplication.infrastructure.plugins.ILogReplicationVersionAdapter;
+import org.corfudb.infrastructure.logreplication.infrastructure.plugins.LogReplicationPluginConfig;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.CorfuStoreEntry;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.exceptions.TransactionAbortedException;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+import org.corfudb.util.retry.IRetry;
+import org.corfudb.util.retry.IntervalRetry;
+import org.corfudb.util.retry.RetryNeededException;
+import org.corfudb.utils.CommonTypes.Uuid;
+import org.corfudb.utils.LogReplicationStreams.Version;
+import org.corfudb.utils.LogReplicationStreams.VersionString;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.NoSuchElementException;
+
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+@Slf4j
+public class UpgradeManager {
+    public static final String LOG_REPLICATION_PLUGIN_VERSION_TABLE = "LogReplicationPluginVersion";
+    public static final String VERSION_PLUGIN_KEY = "VERSION";
+
+    private static final Uuid defaultMetadata = Uuid.newBuilder().setLsb(0).setMsb(0).build();
```

Comment:
why do we even need this default value? I'm thinking actually the metadata value type is incorrect, right? that we might also clean when we do the proper cleanup.

Reply:
Yes, metadata schema need not be of Uuid type.  Adding a TODO for it.

Also removed defaultMetadata and setting the field to null now.